packages/fast-components-react-msft/src/slider-label/slider-label.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    SliderLabelHandledProps,
+    SliderLabelProps,
+    SliderLabelUnhandledProps,
+} from "./slider-label.props";
+import {
+    SliderContext,
+    SliderContextType,
+    SliderTrackItem as BaseSliderTrackItem,
+    SliderTrackItemAnchor,
+} from "@microsoft/fast-components-react-base";
+import { get } from "lodash-es";
+import { DisplayNamePrefix } from "../utilities";
+
+class SliderLabel extends Foundation<
```

Comment:
What is the value of having the slider label as a separate component?

Reply:
I think it is a recognizable pattern to pass labels as children.  I thought about a datalist (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range) approach where the slider component was more opaque about creating labels, but having them be independent components seemed more flexible and not more complex to understand than a "datalist".  All of the items that are value bound on a slider track are composed with SliderTrackItem - the thumb, the labels and the bars.  With this approach authors can easily a wide range of slider decorations - custom labels, bars that highlight particular regions, etc... using the same approach.  They could even mix custom elements with our canned versions.